### PR TITLE
The Studio: Support setting an interval to show random recs

### PR DESCRIPTION
**Pull request by mia-pi-git**
@KrisXV asked me to do this.

### CODE REVIEW

File: server/chat-plugins/the-studio.ts [hunk 1]
```diff
@@ -213,6 +221,28 @@ class RecommendationsInterface {
...
 
+	runInterval(time: string) {
+		let interval = Number(time);
+		if (isNaN(interval)) throw new Chat.ErrorMessage(`Invalid interval time.`);
+		if (interval >= Chat.MAX_TIMEOUT_DURATION) {
+			throw new Chat.ErrorMessage('That interval is too long. Please choose a lower number.');
+		}
+		this.intervalTime = interval;
+		recommendations.intervalTime = interval;
+		interval = interval * 60 * 1000;
```
Comment (KrisXV) [inline server/chat-plugins/the-studio.ts:232]: It should default to once per hour

File: server/chat-plugins/the-studio.ts [hunk 2]
```diff
@@ -213,6 +221,28 @@ class RecommendationsInterface {
...
+		saveRecommendations();
+		if (this.interval) clearInterval(this.interval);
+		this.interval = setInterval(() => {
+			void (async () => {
+				const room = Rooms.get('thestudio');
+				if (!room) return; // do nothing if the room doesn't exist anymore
+				const res = await this.render(this.getRandomRecommendation());
+				room.add(`|html|<div class="infobox">${res}</div>`).update();
```
Comment (KrisXV) [inline server/chat-plugins/the-studio.ts:240]: No info box around it pls

**Comment by Zarel** [comment]
I'm honestly getting really wary of all these circumventions of the "don't spam the user" rule.

I think rooms should have a single `room.interval` which is shared between repeats, this plugin, and any other plugin that uses timers. If one of them is set, the others are not allowed to be set.

**Comment by KrisXV** [comment]
hello @mia-pi-git can u fix conflicts on this por favor

**Pull request closed without merging**